edgelet/provisioning/src/provisioning.rs

```diff
@@ -853,7 +866,16 @@ where
     }
 
     fn reprovision(&self) -> Box<dyn Future<Item = (), Error = Error> + Send> {
-        panic!("A reprovisioning operation is not expected for `BackupProvisioning`")
+        // We've been told to reprovision, so whatever provisioning info we have in the backup should be discarded
```

Comment:
To be conservative, how about we still keep the backup? 

One possibility is that customers switch from true to false. Or some offline scenario that we want things continue working?

Reply:
As discussed in yesterday's scrum, deleting the backup is the signal that we need to reprovision on next startup.